tutorials/how_to_dataloader.ipynb

```diff
@@ -0,0 +1,420 @@
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/steve/NMRcraft/.venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Seed set to 42.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from nmrcraft.data.dataset import DataLoader\n",
+    "\n",
+    "feature_columns = [\n",
+    "            \"M_sigma11_ppm\",\n",
+    "            \"M_sigma22_ppm\",\n",
+    "            \"M_sigma33_ppm\",\n",
+    "            \"E_sigma11_ppm\",\n",
+    "            \"E_sigma22_ppm\",\n",
+    "            \"E_sigma33_ppm\",\n",
+    "        ]\n",
+    "\n",
+    "data_loader = DataLoader(\n",
+    "    feature_columns=feature_columns,\n",
+    "    target_columns=\"metal\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "But of course there are a lot of args to choose from including these following that are meant to be chosen by you guys"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nmrcraft.data.dataset import DataLoader\n",
+    "\n",
+    "feature_columns = [\n",
+    "            \"M_sigma11_ppm\",\n",
+    "            \"M_sigma22_ppm\",\n",
+    "            \"M_sigma33_ppm\",\n",
+    "            \"E_sigma11_ppm\",\n",
+    "            \"E_sigma22_ppm\",\n",
+    "            \"E_sigma33_ppm\",\n",
+    "        ]\n",
+    "\n",
+    "data_loader = DataLoader(\n",
+    "    feature_columns=feature_columns,\n",
+    "    target_columns=\"metal_X4_E\",\n",
+    "    test_size=0.2,\n",
+    "    random_state=42,\n",
+    "    dataset_size=0.001,\n",
+    "    target_type=\"categorical\",  # can be \"categorical\" or \"one-hot\"\n",
+    "    complex_geometry=\"all\", # can be oct, spy, tbp or all\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load data\n",
+    "To load the dataset you can just call the `load_data()` methodhod. This will return the X and y tests and trains respectively and also the y_labels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train, X_test, y_train, y_test, y_labels = data_loader.load_data()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[-0.2722084  -2.21144124 -1.43206187  3.74585072 -5.16278678 -3.93518974\n",
+      "   0.          0.          1.          0.        ]\n",
```

Comment:
Where do those numbers come from (after the NMR tensors) ? Are those the metal, E, X2, X2, X3 ligands? This is quite hard for me to understand atm

Reply:
Yeah, I'll explain that